### PR TITLE
fix(egg): fallback runtime diagnostic config

### PR DESCRIPTION
**Pull request by killagu**
### Summary
- Fallback runtime diagnostic dumps to `${baseDir}/run` when `config.rundir` is missing.
- Reuse the same rundir fallback for application router dump output.
- Default missing or invalid `workerStartTimeout` to the Egg 10 minute startup timeout.

### Verification
- `pnpm exec vitest run packages/egg/test/egg.test.ts --testNamePattern "runtime diagnostics fallback config"`
- `pnpm --filter egg typecheck`
- `pnpm exec vitest run packages/egg/test/egg.test.ts`

Multica issue: EGG-85

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved runtime directory handling with automatic fallback and ensured creation when missing
  * Enhanced worker startup timeout handling with a sensible default and more robust timeout behavior
* **Tests**
  * Added tests verifying diagnostic dumps are written to the runtime location and startup timeout behavior across edge cases

### CODE REVIEW

File: packages/egg/src/lib/egg.ts [hunk 1]
```diff
@@ -658,10 +664,26 @@ export class EggApplicationCore extends EggCore {
...
     this.ready(() => clearTimeout(startTimeoutTimer));
   }
 
+  protected getRuntimeRundir(): string {
+    const rundir = this.config.rundir;
+    if (typeof rundir === 'string' && rundir.length > 0) {
+      return rundir;
+    }
+    return path.join(this.baseDir, 'run');
+  }
```
[the remark this answers is not in the document]
Comment (killagu) [reply packages/egg/src/lib/egg.ts:670]: The PR body is aligned with the implementation now: it says diagnostics fall back to `${baseDir}/run` when `config.rundir` is missing. No code change needed for this one.

File: packages/egg/src/lib/egg.ts [hunk 2]
```diff
@@ -641,9 +646,10 @@ export class EggApplicationCore extends EggCore {
   }
 
   #setupTimeoutTimer(): void {
+    const workerStartTimeout = this.getWorkerStartTimeout();
     const startTimeoutTimer = setTimeout(() => {
       this.coreLogger.error(this.timing.toString());
-      this.coreLogger.error(`${this.type} still doesn't ready after ${this.config.workerStartTimeout} ms.`);
+      this.coreLogger.error(`${this.type} still doesn't ready after ${workerStartTimeout} ms.`);
```
[the remark this answers is not in the document]
Comment (killagu) [reply packages/egg/src/lib/egg.ts:648]: Fixed in `6d74bddb`: the timeout callback now logs the resolved timing dump file path from `path.join(this.getRuntimeRundir(), ...)`.

File: packages/egg/src/lib/egg.ts [hunk 3]
```diff
@@ -565,11 +567,9 @@ export class EggApplicationCore extends EggCore {
    * @private
    */
   dumpConfig(): void {
-    const rundir = this.config.rundir;
+    const rundir = this.getRuntimeRundir();
```
[the remark this answers is not in the document]
Comment (killagu) [reply packages/egg/src/lib/egg.ts:570]: Fixed in `45e50f6f`: the `dumpConfig()` docstring now describes the resolved `${rundir}` destination instead of hardcoding `run/`.

File: packages/egg/src/lib/application.ts [hunk 4]
```diff
@@ -200,9 +200,10 @@ export class Application extends EggApplicationCore {
     super.dumpConfig();
 
     // dump routers to router.json
-    const rundir = this.config.rundir;
+    const rundir = this.getRuntimeRundir();
```
[the remark this answers is not in the document]
Comment (killagu) [reply packages/egg/src/lib/application.ts:203]: Fixed in `45e50f6f`: the router dump docstring now describes the resolved `${rundir}` destination instead of hardcoding `run/`.

**Comment by killagu** [comment]
Closing this PR because the fallback approach masks the actual bundled-loader failure.

Root cause checked from source: Egg already defines `config.rundir` and `config.workerStartTimeout` in `packages/egg/src/config/config.default.ts`, so seeing them as `undefined` means the framework config load unit was not loaded. In the bundled worker path, Egg framework paths can collapse to the bundle output root / chunk path, so `config.default` from the Egg framework is skipped. The root fix is PR #5930, which preserves framework `eggPaths` and plugin path metadata through the startup manifest and patches generated worker framework paths at runtime.

This PR should not merge as-is; EGG-85 should depend on #5930 / EGG-80 and then re-check the startup logs without adding diagnostic fallbacks.

**Comment by killagu** [comment]
Closing in favor of the root loader-path fix in #5930. The undefined values come from Egg framework config not being loaded in the bundled worker, not from missing diagnostic fallbacks.
